feature/xhr/aggregate/index.js

```diff
@@ -146,19 +151,24 @@ function getPayload (events) {
       nullable(null, numeric, false) // timestamp
     ]
 
+    var insert = '2,'
+
     // add custom attributes
     var attrParts = addCustomAttributes(loader.info.jsAttributes || {}, addString)
     fields.unshift(numeric(attrParts.length))
 
-    payload += fields.join(',')
+    insert += fields.join(',')
 
     if (attrParts && attrParts.length > 0) {
-      payload += ';' + attrParts.join(';')
+      insert += ';' + attrParts.join(';')
     }
 
-    if ((i + 1) < events.length) payload += ';'
-  }
+    if ((i + 1) < events.length) insert += ';'
 
+    // check if the current payload string is too big, if so then increment to a new bucket idx
+    if (exceedsSizeLimit(payload[payloadIdx] + insert)) payload[++payloadIdx] = ver
```

Comment:
I think this will lose the string table for subsequent payloads - strings are included in the payload the first time they appear, and subsequent occurrences are referenced only. You will need to restart the string table for each payload, and definitely should have a test for this.

Reply:
How expensive is it to call `exceedsSizeLimit` for every event? Is it better than checking the final size, and if too big starting from scratch with multiple smaller buckets?

Reply:
i think with this, we are already iterating over the string concat so at minimum it would be O(n) no matter what we do.  Not sure how expensive it is to create a new blob n number of times vs iterating multiple times.  Your above comment however makes me think it's better to just create smaller buckets and start over from scratch because of the referenced string payloads.

Reply:
Also, if pure speed is the name of the game we could also look into using array joins instead of the iterative string concat, but i think at this scale its not going to make a huge difference